awscli/customizations/cloudformation/deployer.py

```diff
@@ -106,6 +108,26 @@ def create_changeset(self, stack_name, cfn_template,
             'Capabilities': capabilities,
             'Description': description,
         }
+
+        # CloudFormation requires template bodies to be less than 51200 bytes
+        # if they are deployed from a file. It supports much larger templates
+        # if the templates exist in s3. If this is a 'big' template, throw it
+        # into s3 first.
+        cfn_template_length = len(cfn_template.encode('utf-8'))
```

Comment:
We need unit tests for this logic. It will be easier if you move this code into a separate method